From b15bb8019094797aee65f35e3e654354b61bfd7b Mon Sep 17 00:00:00 2001
From: Jerone Young <jerone.young@canonical.com>
Date: Fri, 12 Feb 2010 18:08:08 -0600
Subject: [PATCH] Force key release for volume keys on Dell Studio 1557

Dell Studio 1557 does not generate a release code when the volume keys
are pressed, causing them to generate infinite key presses. This forces
key release of these keys.

Signed-off-by: Jerone Young <jerone.young@canonical.com>
Signed-off-by: Martin Pitt <martin.pitt@ubuntu.com>
---
 Makefile.am                                       |    3 ++-
 extras/keymap/95-keyboard-force-release.rules     |    2 ++
 extras/keymap/force-release-maps/dell-studio-1557 |    3 +++
 3 files changed, 7 insertions(+), 1 deletions(-)
 create mode 100755 extras/keymap/force-release-maps/dell-studio-1557

diff --git a/Makefile.am b/Makefile.am
index 069bf7f..1b4e2ea 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -541,7 +541,8 @@ dist_udevkeymap_DATA = \
 
 udevkeymapforcereldir = $(libexecdir)/keymaps/force-release
 dist_udevkeymapforcerel_DATA = \
-	extras/keymap/force-release-maps/samsung-other
+	extras/keymap/force-release-maps/samsung-other \
+	extras/keymap/force-release-maps/dell-studio-1557
 
 extras/keymap/keys.txt: $(INCLUDE_PREFIX)/linux/input.h
 	mkdir -p extras/keymap
diff --git a/extras/keymap/95-keyboard-force-release.rules b/extras/keymap/95-keyboard-force-release.rules
index 689afe9..659ee95 100644
--- a/extras/keymap/95-keyboard-force-release.rules
+++ b/extras/keymap/95-keyboard-force-release.rules
@@ -21,4 +21,6 @@ ENV{DMI_VENDOR}="$attr{[dmi/id]sys_vendor}"
 
 ENV{DMI_VENDOR}=="[sS][aA][mM][sS][uU][nN][gG]*", ATTR{[dmi/id]product_name}=="*N130*|*N140*|*SR70S/SR71S*", RUN+="keyboard-force-release.sh $devpath samsung-other"
 
+ENV{DMI_VENDOR}=="Dell Inc.", ATTR{[dmi/id]product_name}=="Studio 1557", RUN+="keyboard-force-release.sh $devpath dell-studio-1557"
+
 LABEL="force_release_end"
diff --git a/extras/keymap/force-release-maps/dell-studio-1557 b/extras/keymap/force-release-maps/dell-studio-1557
new file mode 100755
index 0000000..1718673
--- /dev/null
+++ b/extras/keymap/force-release-maps/dell-studio-1557
@@ -0,0 +1,3 @@
+0xa0 #mute
+0xae #volume up
+0xb0 #volume down
-- 
1.7.3.2

